pkg/install/stack/seed-mla/stack.go

```diff
@@ -0,0 +1,489 @@
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package seedmla
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+	"slices"
+
+	"github.com/sirupsen/logrus"
+
+	"k8c.io/kubermatic/v2/pkg/install/helm"
+	"k8c.io/kubermatic/v2/pkg/install/stack"
+	"k8c.io/kubermatic/v2/pkg/install/util"
+	"k8c.io/kubermatic/v2/pkg/log"
+
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	MonitoringNamespace    = "monitoring"
+	MonitoringChartsPrefix = "monitoring"
+
+	NodeExporterChartName   = "node-exporter"
+	NodeExporterReleaseName = NodeExporterChartName
+	NodeExporterNamespace   = MonitoringNamespace
+
+	KubeStateMetricsChartName   = "kube-state-metrics"
+	KubeStateMetricsReleaseName = KubeStateMetricsChartName
+	KubeStateMetricsNamespace   = MonitoringNamespace
+
+	GrafanaChartName   = "grafana"
+	GrafanaReleaseName = GrafanaChartName
+	GrafanaNamespace   = MonitoringNamespace
+
+	BlackboxExporterChartName   = "blackbox-exporter"
+	BlackboxExporterReleaseName = BlackboxExporterChartName
+	BlackboxExporterNamespace   = MonitoringNamespace
+
+	AlertManagerChartName   = "alertmanager"
+	AlertManagerReleaseName = AlertManagerChartName
+	AlertManagerNamespace   = MonitoringNamespace
+
+	PrometheusChartName   = "prometheus"
+	PrometheusReleaseName = PrometheusChartName
+	PrometheusNamespace   = MonitoringNamespace
+
+	HelmExporterChartName   = "helm-exporter"
+	HelmExporterReleaseName = HelmExporterChartName
+	HelmExporterNamespace   = MonitoringNamespace
+
+	KarmaChartName   = "karma"
+	KarmaReleaseName = KarmaChartName
+	KarmaNamespace   = MonitoringNamespace
+
+	MonitoringIAPChartName = "iap"
+	IAPReleaseName         = MonitoringIAPChartName
+	IAPNamespace           = MonitoringNamespace
+
+	LoggingNamespace    = "logging"
+	LoggingChartsPrefix = "logging"
+
+	LokiChartName   = "loki"
+	LokiReleaseName = LokiChartName
+	LokiNamespace   = LoggingNamespace
+
+	PromtailChartName   = "promtail"
+	PromtailReleaseName = PromtailChartName
+	PromtailNamespace   = MonitoringNamespace
+)
+
+type Monitoring struct{}
+
+func NewStack() stack.Stack {
+	return &Monitoring{}
+}
+
+var _ stack.Stack = &Monitoring{}
+
+func (*Monitoring) Name() string {
+	return "KKP Seed MLA Stack"
+}
+
+func (s *Monitoring) Deploy(ctx context.Context, opt stack.DeployOptions) error {
+	if err := deployNodeExporter(ctx, opt.Logger, opt.KubeClient, opt.HelmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy Node Exporter: %w", err)
+	}
+
+	if err := deployKubeStateMetrics(ctx, opt.Logger, opt.KubeClient, opt.HelmClient, opt); err != nil {
+		return fmt.Errorf("failed to deploy Kube State Metrics: %w", err)
```

Comment:
This is more commonly known as "kube-state-metrics"

Reply:
Done